variables.tf

```diff
@@ -14,12 +14,61 @@
  * limitations under the License.
  */
 
-variable "project_id" {
-  description = "The project ID to deploy to"
+variable "tag_for" {
   type        = string
+  description = "Tags Created for organization or project"
+  default     = "organization"
 }
 
-variable "bucket_name" {
-  description = "The name of the bucket to create"
+variable "org_id" {
   type        = string
+  description = "Organization ID"
+  default     = ""
+}
+
+variable "project_number" {
+  type        = string
+  description = "Project Number"
+  default     = ""
+}
+
+variable "key" {
+  description = "Key for Tags"
+  type        = string
+}
+
+variable "key_description" {
+  type        = string
+  description = "Description for the Key"
+  default     = ""
+}
+
+variable "key_purpose" {
+  type        = string
+  description = "Purpose specification for the Key"
+  default     = ""
+}
+
+variable "key_purpose_data" {
+  type        = map(string)
+  description = "Purpose Data specification for the key"
+  default     = {}
+}
+
+variable "key_iam" {
+  type        = map(list(string))
+  description = "Additive IAM members for the Key"
+  default     = {}
+}
+
+variable "values" {
```

Comment:
Can you rename this as `value_specs` or something else since you already have an attribute called `value` in the object definition which can be mistaken as the variable itself when reading the code.